src/System.Runtime.Serialization.Xml/tests/XmlDictionaryWriterTest.cs

```diff
@@ -107,7 +107,8 @@ public static void XmlBaseWriter_CheckAsync_ThrowInvalidOperationException()
             var t2 = Assert.ThrowsAsync<InvalidOperationException>(() => writer.WriteBase64Async(bytes, 0, byteSize));
 
             InvalidOperationException e = t2.Result;
-            Assert.StrictEqual(e.Message, "An asynchronous operation is already in progress.");
+            bool isAsyncIsRunningException = e.Message.Contains("XmlAsyncIsRunningException") || e.Message.Contains("in progress");
```

Comment:
I think checking with the string in resx file is the right way so we should do it for NetNative and Coreclr as well to make sure it doesn't fail with localized string but this may be not be a priority.